viewer/src/datamodels/cad/sector/rxSectorUtilities.ts

```diff
@@ -14,15 +14,18 @@ import { filter, switchMap, tap, publish, subscribeOn } from 'rxjs/operators';
 export function handleDetermineSectorsInput(
   sectorRepository: Repository,
   sectorCuller: SectorCuller,
-  modelStateHandler: ModelStateHandler
+  modelStateHandler: ModelStateHandler,
+  spendageReportCb: (spendage: SectorLoadingSpendage) => void
 ): OperatorFunction<DetermineSectorsInput, ConsumedSector> {
   return publish((source$: Observable<DetermineSectorsInput>) => {
     const updateSector = (input: DetermineSectorsInput) => {
       const { cameraInMotion } = input;
       if (cameraInMotion) {
         return EMPTY;
       }
-      return from(sectorCuller.determineSectors(input)).pipe(
+      const prioritizedResult = sectorCuller.determineSectors(input);
+      spendageReportCb(prioritizedResult.spendage);
```

Comment:
Again, don't understand Cb, does this generate a report, add data to a report?

Reply:
Tbh, the callback is horrible. I just got tired of fighting RxJS to make a "function" that returns a list of nodes + some statistics in some sensible way.

Renamed the callback.